### PR TITLE
Port backend to shared database package

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-5908

> Just by looking at the count of lines changed I'm getting exhausted BUT:
- mostly copypasta covered by test scenarios
- mostly copypasta covered by test scenarios
- mostly copypasta covered by test scenarios

## Stage 1 migration out of 3/4 stages of DAL tooling migration
- Two versions of repositories exist
- All ported to kysely + Prisma equivalents
- Not used in the production code yet
- Compatibility check assured by dual-suite repositories tests - each repository test now covers both legacy and new repository
- Altering one repository makes you alter a new one as well to ensure consistency throughout the migration process

### CODE REVIEW

File: packages/database/src/index.ts [hunk 1]
```diff
@@ -29,6 +53,31 @@ export function createRepositories(config?: PoolConfig) {
     tokenMeta: new TokenMetaRepository(db),
     token: new TokenRepository(db),
     cache: new CacheRepository(db),
+
+    // L2BEAT - public
+    aggregatedL2Cost: new AggregatedL2CostRepository(db),
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/database/src/index.ts:58]: https://github.com/l2beat/l2beat/pull/4208/commits/671108ad6a7f0d15b93a45663e80db8a6c75a94e

File: packages/database/src/activity-starkex/entity.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { ProjectId, UnixTime } from '@l2beat/shared-pure'
+import { Insertable, Selectable } from 'kysely'
+import { Starkex as StarkexTransactionCountRow } from '../kysely/generated/types'
+
+export interface StarkexTransactionCount {
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/database/src/activity-starkex/entity.ts]: https://github.com/l2beat/l2beat/pull/4208/commits/671108ad6a7f0d15b93a45663e80db8a6c75a94e

File: packages/database/src/activity-block/repository.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    const scope = trx ?? this.db
+    for (const record of records) {
+      await this.addOrUpdate(record, scope)
+    }
+    return records.length
+  }
+
+  async addOrUpdate(
+    record: BlockTransactionCount,
+    trx?: Transaction | PostgresDatabase,
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/database/src/activity-block/repository.ts]: As discussed, we can think about that

File: packages/database/src/acitivity-zksync/repository.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { UnixTime } from '@l2beat/shared-pure'
+import { PostgresDatabase, Transaction } from '../kysely'
+import { ZksyncTransaction, toRecord, toRow } from './entity'
+import { selectZksyncTransaction } from './select'
+
+export class ZksyncTransactionRepository {
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/database/src/acitivity-zksync/repository.ts]: https://github.com/l2beat/l2beat/pull/4208/commits/671108ad6a7f0d15b93a45663e80db8a6c75a94e

File: packages/backend/src/tools/uif/IndexerConfigurationRepository.test.ts [hunk 5]
```diff
@@ -7,117 +7,124 @@ import {
   IndexerConfigurationRepository,
 } from './IndexerConfigurationRepository'
 
-describeDatabase(IndexerConfigurationRepository.name, (database) => {
-  const repository = new IndexerConfigurationRepository(database, Logger.SILENT)
+describeDatabase(IndexerConfigurationRepository.name, (knex, kysely) => {
+  const oldRepo = new IndexerConfigurationRepository(knex, Logger.SILENT)
+  const newRepo = kysely.indexerConfiguration
+
+  // suite(oldRepo)
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/tools/uif/IndexerConfigurationRepository.test.ts]: Done

**Review by imxeno — APPROVED** [review]

**Pull request merged**